main.py

```diff
@@ -42,8 +107,47 @@ def api_label(my_path=None):
         global UPLOAD_LIMIT
         if UPLOAD_LIMIT is not None and CURRENT_UPLOAD_SIZE >= UPLOAD_LIMIT:
             return Response(status=400, response='Reached upload limit')
+        check = os.path.isfile(os.path.join('static/data/', (my_path + '.png')))
+        if check == False:
+            return Response(status=400, response='Image does not exist.')
         with open(os.path.join('static/data/', (my_path + '.json')), "w+") as outfile:
-            json.dump(request.get_json(), outfile, sort_keys=True, indent=4)
+            data = request.get_json()
+            json.dump(data, outfile, sort_keys=True, indent=4)
```

Comment:
You should verify the json before saving it (like you do with the template route). The way it is now it looks like it will save invalid json to the disk and return it next time a GET request comes in. Plus this would avoid having to write then read the same file.